spyql/processor.py

```diff
@@ -44,38 +45,56 @@ def init_vars():
         spyql.log.user_debug(f"Init file not found: {init_fname}")
     except Exception as e:
         spyql.log.user_warning(f"Could not load {init_fname}", e)
+    finally:
+        # user defined variables
+        vars["user_query_vars"] = user_query_vars
     return vars
 
 
 class Processor:
     @staticmethod
-    def make_processor(prs, strings, input_options):
+    def make_processor(prs, strings, interactive = False, input_options = {}):
         """
         Factory for making a file processor based on the parsed query
         """
+
+        spyql.log.user_info(f"Input Options: {input_options}")
+
+        # Since I did not want to damage the existing flow using FROM, when the query
+        # is *really* interactive (as determined by Q), we set "interactive" in the
+        # parsed dict. This is different from interactive passed in the spec above,
+        # which tells the processors that you have to read data from filepath and not
+        # from stdin.
+        if prs.get("interactive", None):
+            return InteractiveProcessor(prs=prs, strings=strings, **input_options)
+
         try:
+
             processor_name = prs["from"]
             if not processor_name:
-                return Processor(prs, strings, **input_options)
+                return Processor(prs=prs, strings=strings, interactive=interactive, **input_options)
 
             processor_name = processor_name.upper()
 
             if processor_name == "JSON":
-                return JSONProcessor(prs, strings, **input_options)
-            if processor_name == "CSV":
-                return CSVProcessor(prs, strings, **input_options)
-            if processor_name == "TEXT":  # single col
-                return TextProcessor(prs, strings, **input_options)
-            if processor_name == "SPY":
-                return SpyProcessor(prs, strings, **input_options)
-
-            return PythonExprProcessor(prs, strings, **input_options)
+                return JSONProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+            elif processor_name == "CSV":
+                return CSVProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+            elif processor_name == "TEXT":  # single col
+                return TextProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+            elif processor_name == "SPY":
+                return SpyProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+
+            return PythonExprProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
```

Comment:
I would revert  this change.
`elif`s are just a matter of style here since if the previous `if` holds true, then the method returns. So the `else` is not doing much... I prefer using `if`s in this case, but not sure if there are any styling guidelines about this. Maybe aiming for readability would be best...

Reply:
I read the Google's [styleguide](https://google.github.io/styleguide/pyguide.html), there's nothing mentioned. Anyways I prefer using `elif`s because when the code AST is generated the entire conditional block looks like a single operation, whereas in case of bunch of `if`s, each step is treated like a different operations.

![Screenshot 2022-02-20 at 12 05 22 PM](https://user-images.githubusercontent.com/18363268/154831575-fa74c819-735a-4383-9db6-e8e38e8c11ae.png)

vs.

![Screenshot 2022-02-20 at 12 05 51 PM](https://user-images.githubusercontent.com/18363268/154831586-63a9ab88-9722-4e7f-8e4c-dcbdfbf46ac8.png)

But I'll leave it to you, whatever you prefer, for now I am leaving it as elifs.

Reply:
Makes sense leaving as `elif`s since we are emulating a `switch ... case`, which would be a single expression.